Frontend/RecRock-Calendar-Frontend/src/app/calendar/components/login/login.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UserService } from '@redrock/services/user.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+})
+export class LoginComponent implements OnInit {
+  rememberMe: boolean = false;
+
+  loginForm: FormGroup = new FormGroup({usernameFormControl: new FormControl('', [Validators.required]), passwordFormControl: new FormControl('', [
+    Validators.required,
+    Validators.minLength(8),
```

Comment:
add a custom validator to check if the password contains letters, numbers, and also special characters (this is a good exercise to learn custom validators). See: https://angular.io/guide/form-validation#defining-custom-validators